packages/manager/modules/vps/src/dashboard/cancel-resiliation/vps-cancel-resiliation.routing.js

```diff
@@ -0,0 +1,76 @@
+import snakeCase from 'lodash/snakeCase';
+import { BillingService } from '@ovh-ux/manager-models';
+
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('vps.detail.dashboard.cancel-resiliation', {
+    url: '/cancel-resiliation',
+    views: {
+      modal: {
+        component: 'billingCancelResiliation',
+      },
+    },
+    layout: 'modal',
+    resolve: {
+      cancelResiliation: /* @ngInject */ (
+        $http,
+        $q,
+        engagement,
+        setReactivateEngagementStrategy,
+      ) => (service) => {
+        if (engagement) {
+          return setReactivateEngagementStrategy();
+        }
+        service.cancelResiliation();
+        return $http
+          .put(`/vps/${service.domain}/serviceInfos`, {
+            renew: service.renew,
+          })
+          .then((result) => {
+            if (result.status === 200) {
+              return result;
+            }
+            return $q.reject(result);
+          });
```

Comment:
Cannot use `catch` ?

Reply:
The error is being handled in the `billingCancelResiliation` component. Hence we will have to reject the promise in order for the error handling to work properly. Thanks :)